src/include/txn_inline.h

```diff
@@ -619,7 +627,7 @@ __wt_txn_op_set_timestamp(WT_SESSION_IMPL *session, WT_TXN_OP *op, bool validate
     if (!__txn_should_assign_timestamp(session, op)) {
         if (validate) {
             if (op->type == WT_TXN_OP_REF_DELETE)
-                WT_RET(__wt_txn_op_delete_commit_apply_timestamps(session, op, validate));
+                WT_RET(__wt_txn_op_delete_commit_apply_timestamps(session, op, validate, false));
```

Comment:
This reads to me as though we should only be doing a timestamp validation check here yet the function name implies "apply timestamp" which is then negated by the new boolean. Should the timestamp check inside `__wt_txn_op_delete_commit_apply_timestamps` instead be refactored out so that only the validation check can be called?

Reply:
Yes, I understand your point. We need to lock the ref and perform the validation check. To assign the timestamp we need to lock the same ref. Due to this reason, Within one time lock itself, we can perform both the validation and assignment.

Any better function name to reduce this ambiguity?

Reply:
I see, that makese sense for optimisation. It's already a very long name so I don't think we can add any more description to it. `__wt_txn_op_delete_commit_validate_and_apply_timestamps` is much too long. I would either add a comment above `if (validate) {` or add to the function `__wt_txn_op_delete_commit_apply_timestamps` comment.

Reply:
can we just leave the function as `__wt_txn_op_delete_commit`?

Reply:
Thanks, I went with suggested function name `__wt_txn_op_delete_commit`